maintainer_guide.md

```diff
@@ -0,0 +1,75 @@
+## Torchvision maintainers guide
+
+This document aims at documenting user-facing policies / principles used when
+developing and maintaining torchvision. Other maintainer info (e.g. release
+process) can be found in the internal wiki.
+
+### What is public and what is private?
+
+For the Python API, torchvision largely follows the [torch core
+policy](https://github.com/pytorch/pytorch/wiki/Public-API-definition-and-documentation)
+which is consistent with other major packages (numpy, scikit-learn etc.).
+We recognize that his policy is somewhat imperfect for some edge cases, and that
+it's difficult to come up with an accurate technical definition. In broad terms,
+which are usually well understood by user, the policy is that:
+
+- modules that can be accessed without leading underscore are public
+- objects in a public file that don't have a leading underscore are public
+- objects that start with a leading underscore are private unless they're
+  exposed / aliased as public in a public `__init__.py` file
```

Comment:
Are you referring to the situation discussed offline, i.e. defining public function in a private namespace and expose them in a public one? If so, can we clarify the text? The "exposed / aliased" implies for me that if we expose `_foo` in a public `__init__.py`, it can be considered public. Was that your intention?

Reply:
Happy to clarify, I tried "exposed / aliased **as public**" but maybe it's not clear enough. Perhaps

> exposed / aliased **as public** (i.e. without a leading underscore) in a public `__init__.py` file

?

Reply:
Maybe a simpler way is to just say:

> Any object exposed in a public `__init__.py` is public unless it has a leading underscore in its immediate name

?

Reply:
Yes, I like that a lot better

Reply:
So it's just a special case of what we have above:

> - objects in a public file that don't have a leading underscore are public

Do you think it's worth re-iterating that with `__init__.py` ?

Reply:
Technically yes, since `__init__.py` itself is not a public file given its underscores. But since have stated above that our definition might technically not be perfect and this should be common knowledge for Python devs, I'm ok to merge the two.